arch/xtensa/src/esp32/esp32_wlan.c

```diff
@@ -0,0 +1,1589 @@
+ *   handling or from watchdog based polling.
+ *
+ * Input Parameters:
+ *   priv - Reference to the driver state structure
+ *
+ * Returned Value:
+ *   OK on success; a negated errno on failure
+ *
+ * Assumptions:
+ *   May or may not be called from an interrupt handler.  In either case,
+ *   the network is locked.
+ *
+ ****************************************************************************/
+
+static int esp_transmit(FAR struct esp_dev_s *priv)
+{
+  int ret = 0;
+  uint8_t *buffer;
+  uint32_t buffer_len;
+
+  /* Set up all but the last TX descriptor */
+
+  buffer = priv->esp_dev.d_buf;
+  buffer_len = priv->esp_dev.d_len;
+  ret = esp_wifi_sta_send_data(buffer, buffer_len);
+
+  if (ret != 0)
+    {
+      wlerr("ERROR: Failed to transmit frame\n");
+      (void)wd_start(&priv->esp_txtimeout, ESP_TXTIMEOUT,
+                     esp_txtimeout_expiry, (uint32_t)priv);
+      return -EIO;
+    }
+
+  return OK;
+}
+
+/****************************************************************************
+ * Function: esp_recvframe
+ *
+ * Description:
+ *   The function is called when a frame is received using the DMA receive
+ *   interrupt.  It scans the RX descriptors of the received frame.
+ *
+ *   NOTE: This function will silently discard any packets containing errors.
+ *
+ * Input Parameters:
+ *   priv  - Reference to the driver state structure
+ *
+ * Returned Value:
+ *   OK if a packet was successfully returned; -EAGAIN if there are no
+ *   further packets available
+ *
+ * Assumptions:
+ *   Global interrupts are disabled by interrupt handling logic.
+ *
+ ****************************************************************************/
+
+static int esp_recvframe(FAR struct esp_dev_s *priv)
+{
+  struct net_driver_s *dev = &priv->esp_dev;
+  uint8_t *buffer;
+  uint32_t buffer_len = 0;
+  buffer = dev->d_buf;
+  buffer_len = dev->d_len;
+
+  /* Check if there are free buffers.  We cannot receive new frames in this
+   * design unless there is at least one free buffer.
+   */
+
+  if (!esp_isfreebuffer(priv))
+    {
+      wlerr("ERROR: No free buffers\n");
+      return -ENOMEM;
+    }
+
+  /* Check if any errors are reported in the frame */
+
+  if (buffer == NULL || buffer_len == 0)
+    {
+      return -EAGAIN;
+    }
+
+  return OK;
+}
+
+/****************************************************************************
+ * Function: esp_receive
+ *
+ * Description:
+ *   An interrupt was received indicating the availability of a new RX packet
+ *
+ * Input Parameters:
+ *   priv  - Reference to the driver state structure
+ *
+ * Returned Value:
+ *   None
+ *
+ * Assumptions:
+ *   Global interrupts are disabled by interrupt handling logic.
+ *
+ ****************************************************************************/
+
+static void esp_receive(FAR struct esp_dev_s *priv)
+{
+  struct net_driver_s *dev = &priv->esp_dev;
+
+  /* Loop while while stm32_recvframe() successfully retrieves valid
```

Comment:
Reference to wrong function name.
```suggestion
  /* Loop while while esp_recvframe() successfully retrieves valid
```

Reply:
Done.